src/main/java/eu/dissco/core/digitalspecimenprocessor/service/MidsService.java

```diff
@@ -54,88 +57,175 @@ private boolean compliesToMidsTwo(DigitalSpecimen attributes) {
+      }
+    }
+    return hasScientificNameId && hasIdentifiedById;
+  }
+
   private boolean compliesToMidsTwoBio(DigitalSpecimen attributes) {
-    return (attributes.getOdsMarkedAsType() != null && attributes.getOdsMarkedAsType())
+    return attributes.getOdsMarkedAsType() != null
         && (attributes.getOdsHasMedia() != null && attributes.getOdsHasMedia())
         && isQualitativeLocationValid(attributes)
         && isQuantitativeLocationValid(attributes)
-        && (occurrenceIsPresent(attributes) && isValid(
-        (attributes.getOccurrences().get(0).getDwcEventDate())))
-        && (occurrenceIsPresent(attributes) && isValid(
-        attributes.getOccurrences().get(0).getDwcFieldNumber()))
-        && isValid(attributes.getDwcRecordedBy());
+        && (occurrenceIsPresent(attributes) && hasValue(
+        attributes.getDwcOccurrence().get(0).getDwcEventDate(),
+        attributes.getDwcOccurrence().get(0).getDwcVerbatimEventDate(),
+        convertInteger(attributes.getDwcOccurrence().get(0).getDwcYear())))
+        && (occurrenceIsPresent(attributes) &&
+        hasValue(attributes.getDwcOccurrence().get(0).getDwcFieldNumber(),
+            attributes.getDwcOccurrence().get(0).getDwcRecordNumber()))
+        && hasValue(attributes.getDwcRecordedBy(), attributes.getDwcRecordedById());
+  }
+
+  private String convertInteger(Integer integer) {
+    if (integer != null) {
+      return integer.toString();
+    } else {
+      return null;
+    }
+  }
+
+  private boolean hasValue(String... terms) {
+    for (var term : terms) {
+      if (isValid(term)) {
+        return true;
+      }
+    }
+    return false;
   }
 
 
   private boolean compliesToMidsTwoPaleoBio(DigitalSpecimen attributes) {
-    return (attributes.getOdsMarkedAsType() != null && attributes.getOdsMarkedAsType())
+    return attributes.getOdsMarkedAsType() != null
         && isStratigraphyValid(attributes)
         && isQualitativeLocationValid(attributes)
         && isQuantitativeLocationValid(attributes);
   }
 
   private boolean isQuantitativeLocationValid(DigitalSpecimen digitalSpecimen) {
     if (locationIsPresent(digitalSpecimen)
-        && digitalSpecimen.getOccurrences().get(0).getLocation().getGeoreference() != null) {
-      var georeference = digitalSpecimen.getOccurrences().get(0).getLocation().getGeoreference();
-      return georeference.getDwcDecimalLatitude() != null
-          && georeference.getDwcDecimalLongitude() != null;
+        && digitalSpecimen.getDwcOccurrence().get(0).getDctermsLocation().getGeoReference()
+        != null) {
+      var location = digitalSpecimen.getDwcOccurrence().get(0).getDctermsLocation();
+      var georeference = location.getGeoReference();
+      var validValue = hasValue(location.getDwcLocationID(), georeference.getDwcFootprintWkt());
+      if (!validValue) {
+        validValue = georeference.getDwcDecimalLatitude() != null
+            && georeference.getDwcDecimalLongitude() != null;
+      }
```

Comment:
this "validValue" param is confusing, simpler to do:

```
return hasValue(location.getDwcLocationID(), georeference.getDwcFootprintWkt()) ||
 validValue = georeference.getDwcDecimalLatitude() != null
            && georeference.getDwcDecimalLongitude() != null;
```

Reply:
Compounded it in one statement